de/factorization.py

```diff
@@ -12,14 +12,29 @@
 
 def calcW(data, eta, alphaIn):
     """Directly calculate w."""
-    U = np.copy(data)
-    np.fill_diagonal(U, 0.0)
 
-    B = (data * alphaIn) / eta[:, np.newaxis]
-    assert np.all(np.isfinite(B))
-    B = logit(np.clip(B, 0.0001, 0.9999))
-    assert np.all(np.isfinite(B))
-    return np.linalg.lstsq(U.T, B.T, rcond=None)[0].T
+    w = None
+
+    
+    if (all(isinstance(i, np.ndarray) for i in data)):
+      for x in data:
+        U1 = np.copy(x)
+        np.fill_diagonal(U1, 0.0)
+        B = (x * alphaIn) / eta[:, np.newaxis]
+        assert np.all(np.isfinite(B))
```

Comment:
in line 23, you are using `x` but I guess you should have `U1` since you are filling the diagonal of `U1` with zeros...

Reply:
Try that let me know if you still have the problem.

Reply:
The problem still exists.